CRM/Core/BAO/Domain.php

```diff
@@ -366,7 +366,11 @@ public static function getDefaultReceiptFrom() {
    */
   public static function getNoReplyEmailAddress() {
     $emailDomain = CRM_Core_BAO_MailSettings::defaultDomain();
-    return "do-not-reply@$emailDomain";
+    $noReplyAddress = \Civi\Api4\Setting::get(FALSE)
+      ->addSelect('no_reply_email_address:label')
+      ->execute()->first();
```

Comment:
With no pseudoconstant this is even simpler:
```suggestion
    $noReplyAddress = Civi::settings()->get('no_reply_email_address');
```

Reply:
PR updated, but with the Email Rule, the user can not save an empty 'no reply email address'. @colemanw